guides/plugins/plugins/content/cms/real-world-cms/add-real-world-cms-element.md

```diff
@@ -0,0 +1,460 @@
+
+## Prerequisites
+
+We assume that you know the basics for about creating a CMS-Element from [Creating a custom CMS element](../add-cms-element.md). So if you don't understand
+something please make sure to take a look at that guide as well.
+
+## Creating your custom CMS element
+
+Imagine you want to create a new element to display a button which is fully configurable by the shop manager. We already build a CMS-Block for the button in the 
+last guide but if you just have a block you don't have a configuration that's why you need a CMS-Element
+
+## Administration Code
+
+Since you already know the basics from [Add CMS Element](../add-cms-element.md), we will just look a quick look at the code.
+If you followed the real world block guide you already should have a `main.js` in the  `<plugin root>/src/Resources/app/administration/src` directory. 
+In this file we need to import our new element's directory. So let's do it:
+
+{% code title="<plugin root>/src/Resources/app/administration/src/main.js" %}
+```javascript
+/* Import the block directory */
+import './module/sw-cms/blocks/text/cms-button';
+
+/* Import the elements directory */
+import './module/sw-cms/elements/cms-button'
+```
+{% endcode %}
+
+Of course we need to create the directory which we just imported now. So let's do that and place an `index.js` in your `<plugin root>/src/Resources/app/administration/src/module/sw-cms/elements/cms-button/index.js`. We need to register our element here and we will keep it pretty simple:
+
+{% code title="<plugin root>/src/Resources/app/administration/src/module/sw-cms/elements/dailymotion/index.js" %}
+```javascript
+import './component';
+import './config';
+import './preview';
+
+
+Shopware.Service('cmsService').registerCmsElement({
+    name: 'cms-button',
+    label: 'sw-cms.blocks.text.ninja-cms-button.label',
+    component: 'sw-cms-el-cms-button',
+    configComponent: 'sw-cms-el-config-cms-button',
+    previewComponent: 'sw-cms-el-preview-cms-button',
+    defaultConfig: {
+        title: {
+            source: 'static',
+            value: 'ButtonText'
+        },
+        textColor: {
+            source: 'static',
+            value: '#fff'
+        },
+        url: {
+            source: 'static',
+            value: ''
+        },
+        newTab: {
+            source: 'static',
+            value: 'true'
+        },
+        buttonAlign: {
+            source: 'static',
+            value: 'center'
+        },
+        buttonColor: {
+            source: 'static',
+            value: '#4492ed'
+        },
+        buttonWidth: {
+            source: 'static',
+            value: ''
+        },
+        buttonHeight: {
+            source: 'static',
+            value: ''
+        }
+    },
+
+});
+```
+{% endcode %}
+
+Of course you could add a bit more configuration but we want to keep it simple in this guide. Now we also need to create the component, preview and the config. Let's start by doing the component.
```

Comment:
8 configurations for a simple button is already a lot. But this is more real life like 👍

Reply:
You actually could add a lot more like: border-radius, border-color, hover-color and so on but I guess 8 config fields is enough for a guide to understand how it works